cmd/agent/api/agent/agent.go

```diff
@@ -251,6 +251,8 @@ func streamLogs(w http.ResponseWriter, r *http.Request) {
 			// The buffer will flush on its own most of the time, but when we run out of logs flush so the client is up to date.
 			flusher.Flush()
 		}
+		// don't hog the CPU
+		time.Sleep(1 * time.Millisecond)
```

Comment:
I think this should only be called when we find no logs in the `logMessageReceiver`, so only in the `else` block above after `flusher.Flush`, otherwise this sleep limits the maximum log throughput that can be streamed by this command to roughly 1000 logs/second.

(and given the design of the `logMessageReceiver`, if this loop doesn't read logs fast enough while the `stream-logs` command runs, its internal buffered channel could end up being full which would apply backpressure upstream on the log processor and block it: https://github.com/DataDog/datadog-agent/blob/0eb90bd73db83b379943f6cba5e15201430c1560/pkg/logs/processor/processor.go#L93. The risk of ending up in this situation sounds low as long as we only sleep for a millisecond when there are no more logs to read)

Reply:
Indeed, moved the sleep in the `else` block.